docker/test

```diff
@@ -59,38 +59,92 @@ while getopts ":c:mn:o:plde" opt; do
   esac
 done
 
-COMPOSE_ARGS="$COMPOSE_ARGS-$OPERATING_SYSTEM.yml"
-
-if ! [ $COMPOSE_ARGS ];
-then
-  echo "Operating system: $OPERATING_SYSTEM not supported!"
-  usage
-  exit 1
-fi
-
 # PROJ_NAME should be set in a jenkins environment. It allows multiple
 #  instances of the compose to run without clobbering each other.
 cd $DIR
-docker-compose $PROJ_NAME stop
-docker-compose $PROJ_NAME rm -f
+
+SERVICE_PORTS_ARG=""
+if [ "$EXPOSE_PORTS" == "1" ]; then
+  SERVICE_PORTS_ARG="-p 8080:8080 -p 8443:8443"
+fi
+
+podman pod rm -f $PROJ_NAME
 
 if [ "$USE_CACHE" != "1" ]; then
-  docker-compose $COMPOSE_ARGS pull
+  podman pull "$REGISTRY/candlepin-base-$OPERATING_SYSTEM"
+  podman pull "$REGISTRY/$DB"
 fi
 
-SERVICE_PORTS_ARG=""
-if [ "$EXPOSE_PORTS" == "1" ]; then
-  SERVICE_PORTS_ARG="--service-ports"
+podman pod create --name $PROJ_NAME
+podman network create network-$PROJ_NAME
+
+ENV_DATABASE=""
+if [[ $DB == postgres* ]]; then
+  ENV_DATABASE="-e USING_POSTGRES=true -e DBHOSTNAME=db"
+  podman create --name="$PROJ_NAME-db" \
+    --pod "$PROJ_NAME" \
+    -e POSTGRES_USER=candlepin \
+    -e POSTGRES_PASSWORD= \
+    -e POSTGRES_DB=candlepin \
+    -e POSTGRES_HOST_AUTH_METHOD=trust \
+    --net network-$PROJ_NAME \
+    --network-alias db \
+    --restart always \
+    $REGISTRY/$DB
+elif [[ $DB == mariadb* ]]; then
+  ENV_DATABASE="-e USING_MYSQL=true -e DBHOSTNAME=db"
+  podman create --name="$PROJ_NAME-db" \
+    --pod "$PROJ_NAME" \
+    -e MYSQL_ROOT_PASSWORD=password \
+    -v ./mysql.cnf:/etc/mysql/conf.d/mysql.cnf \
+    --net network-$PROJ_NAME \
+    --network-alias db \
+    --restart always \
+    $REGISTRY/$DB
 fi
 
+if [[ $OPERATING_SYSTEM == cs8 ]]; then
+  podman create --name="$PROJ_NAME-candlepin" \
+     --pod $PROJ_NAME \
+     $ENV_DATABASE \
+     -v ../:/candlepin-dev \
+     $SERVICE_PORTS_ARG \
+     --net network-$PROJ_NAME \
+     --network-alias candlepin \
+     --privileged \
+     "$REGISTRY/candlepin-base-$OPERATING_SYSTEM"
+elif [[ $OPERATING_SYSTEM == cs9 ]]; then
+  podman create --name="$PROJ_NAME-candlepin" \
+     --pod $PROJ_NAME \
+     $ENV_DATABASE \
+     -v ../:/candlepin-dev \
+     -v /sys/fs/cgroup:/sys/fs/cgroup:ro \
+     $SERVICE_PORTS_ARG \
+     --net network-$PROJ_NAME \
+     --network-alias candlepin \
+     --privileged \
+     "$REGISTRY/candlepin-base-$OPERATING_SYSTEM"
+else
+  echo "We do not support this operating system: $OPERATING_SYSTEM"
+  usage
+  podman pod rm -f $PROJ_NAME
+  podman network rm network-$PROJ_NAME
+  exit 1
+fi
+
+
+podman pod start $PROJ_NAME
+sleep 5 # Waiting to start systemd
+
 if [ "$DETACHED" == "1" ]; then
     # Run detached, and don't remove the containers
-    docker-compose $PROJ_NAME $COMPOSE_ARGS run -d $SERVICE_PORTS_ARG candlepin $TEST_CMD
+    podman exec -d $PROJ_NAME-candlepin $TEST_CMD
     RETVAL=$?
-else
-    docker-compose $PROJ_NAME $COMPOSE_ARGS run --rm $SERVICE_PORTS_ARG candlepin $TEST_CMD
```

Comment:
since we don't use docker-compose*.yml files anymore, we should remove the `docker/switch-defaults` file (it is useless)

Reply:
done